### ISSUE TITLE
Segafults in 4.16.2

### ISSUE THREAD

**Issue by madf** [keshavbhatt/whatsie#212]
The app started crashing periodically.

Stacktrace:
```
Thread 1 "whatsie" received signal SIGSEGV, Segmentation fault.
0x00007fffef93f114 in QWidget::setMinimumWidth(int) () from /usr/lib64/libQt5Widgets.so.5
(gdb) bt
#0  0x00007fffef93f114 in QWidget::setMinimumWidth(int) () at /usr/lib64/libQt5Widgets.so.5
#1  0x00005555555ab915 in operator() (__closure=0x0, notification=std::unique_ptr<QWebEngineNotification> = {...}) at mainwindow.cpp:991
#2  0x00005555555abce9 in std::__invoke_impl<void, MainWindow::setNotificationPresenter(QWebEngineProfile*)::<lambda(std::unique_ptr<QWebEngineNotification>)>&, std::unique_ptr<QWebEngineNotification, std::default_delete<QWebEngineNotification> > > (__f=<optimized out>) at /usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/invoke.h:61
#3  std::__invoke_r<void, MainWindow::setNotificationPresenter(QWebEngineProfile*)::<lambda(std::unique_ptr<QWebEngineNotification>)>&, std::unique_ptr<QWebEngineNotification, std::default_delete<QWebEngineNotification> > >
    (__fn=<optimized out>) at /usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/invoke.h:111
#4  std::_Function_handler<void(std::unique_ptr<QWebEngineNotification, std::default_delete<QWebEngineNotification> >), MainWindow::setNotificationPresenter(QWebEngineProfile*)::<lambda(std::unique_ptr<QWebEngineNotification, std::default_delete<QWebEngineNotification> >)> >::_M_invoke(const std::_Any_data &, std::unique_ptr<QWebEngineNotification, std::default_delete<QWebEngineNotification> > &&) (__functor=<optimized out>, __args#0=<optimized out>)
    at /usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/std_function.h:290
#5  0x00007ffff7dca336 in ??? () at /usr/lib64/libQt5WebEngineWidgets.so.5
#6  0x00007ffff0f81471 in ??? () at /usr/lib64/libQt5WebEngineCore.so.5
#7  0x0000555555c51970 in ??? ()
#8  0x0000555555e135e0 in ??? ()
#9  0x00007fff00000000 in ??? ()
#10 0x0000555556778c60 in ??? ()
#11 0x0000555555c68cb0 in ??? ()
#12 0x0000555555c67ec0 in ??? ()
#13 0x00007fffffffc0c0 in ??? ()
#14 0x0000555555bbd970 in ??? ()
#15 0x0000555555e135e0 in ??? ()
#16 0x0000555555c51970 in ??? ()
#17 0x00007fffffffc060 in ??? ()
#18 0x0000555555bcebb0 in ??? ()
#19 0x00007fffffffc340 in ??? ()
#20 0x00007fffffffc260 in ??? ()
#21 0x00007fffffffc0c0 in ??? ()
#22 0x00007ffff0f811a0 in ??? () at /usr/lib64/libQt5WebEngineCore.so.5
#23 0x00007fffffffc180 in ??? ()
#24 0x00007ffff7019bbf in ??? () at /usr/lib64/libQt5WebEngineCore.so.5
#25 0x00007fff34b15eb0 in ??? ()
#26 0x0000555555cb9540 in ??? ()
#27 0x0000555555d5a2c0 in ??? ()
#28 0x00007fffffffc0a0 in ??? ()
#29 0x00007fffffffc140 in ??? ()
#30 0x0000000000000000 in ??? ()
```

**Comment by keshavbhatt** [comment]
The issue was caused due to Custom notification popup widget life-cycle. It is fixed in [dde6e02](https://github.com/keshavbhatt/whatsie/commit/dde6e024d097bf3fa36a08d2baa59c69f0bc0571)

duplicated by #207

**Comment by shakaran** [comment]
@madf, for start-up segfaults, the maintained fork **Whatly** adds crash recovery that falls back to safer rendering flags after a crash, and is built on current Qt6. https://github.com/shakaran/whatly
